pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -322,8 +322,12 @@ private void takeSnapshot() {
         });
     }
 
-
-
+    private void clearAbortTransaction() {
+        while (!aborts.isEmpty() && !((ManagedLedgerImpl) topic.getManagedLedger())
```

Comment:
IIRC there is an ongoing effort to not use ManafedLedgerImpl but only the interface.
Is it necessary to do this cast?

Reply:
```ledgerExists``` it not the ```ManagedLedger``` interface method.